mirror_builder/resolver.py

```diff
@@ -114,14 +111,15 @@ def get_project_from_pypi(project, extras):
             continue
         name = name_and_version.groups()[0]
         version = name_and_version.groups()[1]
+        is_sdist = name_and_version.groups()[-1] == '.tar.gz'
```

Comment:
A GitHub release tarball is not actually an sdist. They often behave similarly, but the GitHub release is just an archive of everything while an sdist is often more selective about what's included.

If we really care if it's an sdist, we should only set the flag True when downloading from PyPI.

Reply:
Yeah ...

If you take a look at the [2.2.2 release](https://github.com/pytorch/pytorch/releases/tag/v2.2.2) you can see there's two .tar.gz files attached 

What you mean by a "GitHub release tarball" is the `v2.2.2.tar.gz` one. We're downloading the other one

I knew it wasn't just an archive of the git repo, because it's much bigger that the auto-created GitHub one

I've just dug in now to understand the difference ...

What we're downloading is generated by [this "Create Release" job](https://github.com/pytorch/pytorch/actions/runs/8459476685/job/23175854762)

The main difference (at first glance) is down to this:

```
    steps:
      - uses: malfet/checkout@silent-checkout
        with:
          submodules: 'recursive'
```

so the archive contains all of the submodules too (mostly the `third_party/` dir)

Reply:
So ... completely agree we're not downloading an actual sdist for PyTorch, and that needs more thought and investigation, but ...

> If we really care if it's an sdist, we should only set the flag True when downloading from PyPI.

not sure I understand this point ...

In the torch override, I'm doing `resolver.PyPIProvider(only_sdists=False)`

Which basically means "it's ok to resolve this dependency on PyPI with a version that only has wheels" and then we proceed to download that source from elsewhere